components/homme/src/theta-l/model_init_mod.F90

```diff
@@ -62,6 +64,26 @@ subroutine model_init2(elem,hybrid,deriv,hvcoord,tl,nets,nete )
       do t=1,timelevels
          elem(ie)%state%phinh_i(:,:,nlevp,t) = elem(ie)%state%phis(:,:)
       enddo
+
+      ! initialize reference states used by hyberviscosity
+#define HV_REFSTATES_V1
+#ifdef HV_REFSTATES_V0
+      elem(ie)%derived%dp_ref=0
+      elem(ie)%derived%phi_ref=0
+      elem(ie)%derived%theta_ref=0
+#endif
+#ifdef HV_REFSTATES_V1
+      ps_ref(:,:) = hvcoord%ps0 * exp ( -elem(ie)%state%phis(:,:)/(Rgas*TREF)) 
+      do k=1,nlev
+         elem(ie)%derived%dp_ref(:,:,k) = ( hvcoord%hyai(k+1) - hvcoord%hyai(k) )*hvcoord%ps0 + &
```

Comment:
Nightlies show baseline diffs in all SMS theta tests. I replaced prim_advance_mod with the original one and at least 1 test became bfb. Comparing derived% fields with the ones computed in HV routine, dp_ref and phi_ref show diffs. dp_ref is used to compute everything. derived%dp_ref and dp_ref in HV differ in 1e-12, 1e-13 at some levels. I assume  that is where nonbfb diffs come from, but not clear why, lines computing them are identical. Another compiler optimization done differently?

Reply:
Not sure what to do: simplified nonbfb line is  
`     ps_ref(:,:) = elem(ie)%state%phis(:,:)/(Rgas*TREF)`
I put it in dp_ref and in derived%dp_ref; tref, rgas, and phis are the same, but the line above gets e-17 differences. i tried to put rgas*tref into a separate variable, but it did not help. maybe, issue is somewhere else. maybe i should re-run standalone homme with different number of ranks.